src/js/messaging/containers/Main.jsx

```diff
@@ -35,13 +41,66 @@ class Main extends React.Component {
     this.props.setNewFolderName(field);
   }
 
-  handleSubmit() {}
+  handleCreateNewFolderModal() {
+    // Reset the form / state object when closed
+    const resetForm = { value: '', dirty: false };
```

Comment:
Instead of creating the object ourselves, just pass `makeField('')` to `setnewFolderName`?

Also, alternatively, you can also have this within the handler for `TOGGLE_CREATE_FOLDER` so that it resets the state only if it's turning to not visible.